test/mapArray.test.js

```diff
@@ -3,21 +3,15 @@ const { mapArray } = require("../lib");
 describe("mapArray", () => {
   it("returns an array of elements transformed by a callback function", () => {
     const arr = [1,2,3,4,5,6,7,8,9,10];
-    const callback = function(elem) {
-      var result = [];
-      for (var i = 0; i < elem.length; i++) {
-        result.push(elem[i] ** 2);
-      }
-      return result;
-    };
+    const callback = function(elem) { return elem ** 2; };
     expect(mapArray(arr, callback)).toEqual([1,4,9,16,25,36,49,64,81,100]);
   });
 
   it("does not mutate the input array", () => {
-    const arr = [["h","e","l","l","o"], ["h","o","w"], ["a","r","e"], ["y","o","u","?"]];
-    const callback = function(elem) { return elem.reverse(); };
+    const arr = [1,2,3,4,5,6];
+    const callback = function(elem) { return ++elem; };
```

Comment:
I think `++elem` is not a valid mutation test since ++ is an operator and not a member function of the element. See https://repl.it/@mtso/UpsetRelievedCompilerbug

Reply:
@mtso I was surprised to find out that the built in map doesn't even handle your case the way you expect.